### DISCUSSION TITLE
Is there way to do forward and backward prefix scan op for N arrayes  in one call?

### DISCUSSION THREAD

**Discussion by solution3000** [NVIDIA/cccl#1055] [Thrust]
hi，

I have N equal-sized arrayes, e.g., 
   int [N][LEN]  Array; 
and for each &Array[i][0],  firstly I do  forward-direction prefix sum scan, then backward-direction!      

N is a big number(~100000),  and the performance should be taken into consideration！ 
So I need some kind of batched scan functionality in thrust !

Any suggestion is appreciated!  

Thx!

**Comment by tomilov** [comment]
@solution3000 how LEN is big?

**Comment by solution3000** [comment]
> @solution3000 how LEN is big?

LEN=4096, or 8192

**Comment by tomilov** [comment]
I think it better to make 100000 single-threaded scans.

**Comment by solution3000** [comment]
Actually,  datatype of Array is single float(float32), in sum scan,  I must take care of  rounding error, and want to use float-float scheme  to ensure sum scan precision

**Comment by jrhemstad** [comment]
You can just use a `*scan_by_key` algorithm where each sub-array has a different key.
